<launch>
	<arg name="id" default="0"/>
	<arg name="ns" default="$(eval 'robot' + str( arg('id')) )"/>

	<arg name="fcu_url" default="/dev/ttyUSB0:921600" />
	<arg name="gcs_url" default="udp://@192.168.0.171:14550" />
	<arg name="tgt_system" default="1" />
	<arg name="tgt_component" default="1" />

	<include file="$(find formation)/launch/mavros_ns.launch">
		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" default="$(arg gcs_url)" />
		<arg name="tgt_system" default="$(arg tgt_system)" />
		<arg name="tgt_component" default="$(arg tgt_component)" />
		<arg name="log_output" default="screen" />
		<arg name="ns" default="$(arg ns)"/>
	</include>

	<group ns="$(arg ns)">
		<rosparam command="load" file="$(find formation)/config/params.yaml" />
		<param name="myID" type="int" value="$(arg id)" />

		<node pkg="formation" type="robot_node.py" name="robot_node_$(arg id)"  required="true" output="screen">
        </node>
	</group>
	
</launch>
